CustomContent: Add withinDescendant parameter to API

danigm opened the pull request:
## Summary

This patch adds a new optional parameter to the getContentByFilter API call to be able to retrieve all descendant nodes using the MPTT model fields.

With this new parameter it's possible to filter all the nodes below a specific topic node, useful when combining with other filters like kind or tags.

rtibbles commented on packages/hashi/src/kolibri.js:
```diff
@@ -123,6 +130,7 @@ export default class Kolibri extends BaseShim {
        * @param {boolean} options.onlyContent - set to true to query only content nodes
        * @param {boolean} [options.limitToChannel=true] - true to limit the
        * results to the topic channel
+       * @param {MPTTNode} options.withinDescendant - Object to filter nodes descendant of this one
```
This requires that the HTML5 apps are aware of the implementation details of MPTT - perhaps it would be better for this to just be the id of the node to fetch descendants of?

We can add a `descendant_of` filter to the backend: https://github.com/learningequality/kolibri/blob/release-v0.15.x/kolibri/core/content/api.py#L250 that does the lookup for the node - something like:
```
def filter_descendant_of(self, queryset, name, value):
    try:
        node = ContentNode.objects.values("lft", "rght", "tree_id").get(pk=value)
    except (ContentNode.DoesNotExist, ValueError):
        return queryset.none()
    return queryset.filter(lft__gt=node["lft"], rght__lt=node["rght"], tree_id=node["tree_id"])
```

danigm replied:
Okay, that's a better API and moving the logic to the backend simplifies the frontend logic. I'll update the PR to follow that implementation

danigm replied:
I've just updated the PR with the new filter and the parameter `descendantOf` that should be a node id.

danigm committed:
CustomContent: Add descendantOf parameter to API

This patch adds a new optional parameter to the getContentByFilter API
call to be able to retrieve all descendant nodes using the MPTT model
fields.

With this new parameter it's possible to filter all the nodes below a
specific topic node, useful when combining with other filters like kind
or tags.
